test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2138,4 +2142,20 @@ public static <T extends Throwable> T expectThrows(Class<T> expectedType, Action
             () -> builder.get().decRef() // dec ref if we unexpectedly fail to not leak transport response
         );
     }
+
+    public BulkRequestBuilder2 bulkRequestBuilder() {
+        return new BulkRequestBuilder2();
+    }
+
+    public IndexRequestBuilder2 indexRequestBuilder(String index) {
+        return new IndexRequestBuilder2(index);
+    }
+
+    public UpdateRequestBuilder2 updateRequestBuilder(String index, String id) {
+        return new UpdateRequestBuilder2(index, id);
+    }
+
+    public ReindexRequestBuilder2 reindexRequestBuilder(String index) {
+        return new ReindexRequestBuilder2();
+    }
```

Comment:
Ideally we could just replace the `prepareBulk()` etc. methods rather than making new methods and then having to migrate everything over.

Reply:
Yeah that's what I wanted to do, but that makes it harder to do piecemeal, right? And doing this all at once would be impractically large.

Reply:
I'm not sure it's that bad tbh. There'll be other things that need adjusting to add refcounting in this area, but I would expect we can indeed replace today's fake builders with the real builders you're developing in this PR in a single step (i.e. once the APIs line up properly). At least, if we can't do that, it'd be educational to understand why, and maybe we can fix up any obstructions in precursor PRs so that this one goes through nicely.

Reply:
I'm trying a slightly different approach -- modifying the existing builders in place to see what breaks -- #104636.